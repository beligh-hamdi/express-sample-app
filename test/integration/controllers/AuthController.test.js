const app = require('../../../api/server');
const request = require('supertest')(app);
const db = require('../../db');

beforeEach(() => db.clear().then(() => db.init()));

describe(':signup', () => {
  test('Should signup', (done) => {
    request
    .post('/auth/signup')
    .send({ email: 'test1@test.com', password: 'password', firstName: 'name', lastName: 'name' })
    .expect(200)
    .end((err, res) => {
      if (err) fail(err);
      delete res.body.user.password;
      delete res.body.user._id;
      expect(res.body).toMatchSnapshot();
      done();
    });
  });

  test('Should fail signup (no password)', (done) => {
    request
    .post('/auth/signup')
    .send({ email: 'test1@test.com', firstName: 'name', lastName: 'name' })
    .expect(400)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });
});

describe(':login', () => {
  test('Should login', (done) => {
    request
    .post('/auth/login')
    .send({ email: 'test@test.com', password: 'password' })
    .expect(200)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });

  test('Should fail login (bad email)', (done) => {
    request
    .post('/auth/login')
    .send({ email: 'bad@test.com', password: 'password' })
    .expect(401)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });

  test('Should fail login (bad password)', (done) => {
    request
    .post('/auth/login')
    .send({ email: 'test@test.com', password: 'bad' })
    .expect(401)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });

  test('Should fail login (no email)', (done) => {
    request
    .post('/auth/login')
    .send({ password: 'password' })
    .expect(401)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });
});
