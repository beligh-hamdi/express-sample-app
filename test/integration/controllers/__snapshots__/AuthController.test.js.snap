exports[`:login Should fail login (bad email) 1`] = `
Object {
  "errors": Array [
    Object {
      "code": 1,
      "message": "Bad credentials",
    },
  ],
}
`;

exports[`:login Should fail login (bad password) 1`] = `
Object {
  "errors": Array [
    Object {
      "code": 1,
      "message": "Bad credentials",
    },
  ],
}
`;

exports[`:login Should fail login (no email) 1`] = `
Object {
  "errors": Array [
    Object {
      "code": 2,
      "message": "Missing credentials",
    },
  ],
}
`;

exports[`:login Should login 1`] = `
Object {
  "token": "token",
  "user": Object {
    "__v": 0,
    "_id": "589b5b4aa79dff2f7fe6659d",
    "email": "test@test.com",
    "firstName": "name",
    "lastName": "lastname",
  },
}
`;

exports[`:signup Should fail signup (no password) 1`] = `
Object {
  "errors": Array [
    Object {
      "errors": Object {
        "password": Object {
          "kind": "required",
          "message": "User password required",
          "name": "ValidatorError",
          "path": "password",
          "properties": Object {
            "message": "User password required",
            "path": "password",
            "type": "required",
          },
        },
      },
      "message": "User validation failed",
      "name": "ValidationError",
    },
  ],
}
`;

exports[`:signup Should signup 1`] = `
Object {
  "token": "token",
  "user": Object {
    "__v": 0,
    "email": "test1@test.com",
    "firstName": "name",
    "lastName": "name",
  },
}
`;
